src/libm-tester/CMakeLists.txt

```diff
@@ -16,94 +17,31 @@ target_link_libraries(${TARGET_IUT} ${LIB_SLEEF} ${EXTRA_LIBS})
-    PRIVATE ENABLE_AVX=1)
-  target_link_libraries(${TARGET_IUTAVX} ${LIB_SLEEF} ${EXTRA_LIBS})
-
-  list(APPEND IUT_LIST ${TARGET_IUTAVX})
-endif(COMPILER_SUPPORTS_AVX)
-
-if(COMPILER_SUPPORTS_FMA4)
-  add_executable(${TARGET_IUTFMA4} iutsimd.c iutsimdmain.c testerutil.c)
-  target_compile_options(${TARGET_IUTFMA4}
-    PRIVATE ${FLAGS_ENABLE_AVX} ${FLAGS_ENABLE_FMA4})
-  target_compile_definitions(${TARGET_IUTFMA4}
-    PRIVATE ENABLE_FMA4=1)
-  target_link_libraries(${TARGET_IUTFMA4} ${LIB_SLEEF} ${EXTRA_LIBS})
-
-  list(APPEND IUT_LIST ${TARGET_IUTFMA4})
-endif(COMPILER_SUPPORTS_FMA4)
-
-if(COMPILER_SUPPORTS_AVX2)
-  add_executable(${TARGET_IUTAVX2} iutsimd.c iutsimdmain.c testerutil.c)
-  target_compile_options(${TARGET_IUTAVX2}
-    PRIVATE ${FLAGS_ENABLE_AVX2})
-  target_compile_definitions(${TARGET_IUTAVX2}
-    PRIVATE ENABLE_AVX2=1)
-  target_link_libraries(${TARGET_IUTAVX2} ${LIB_SLEEF} ${EXTRA_LIBS})
-
-  list(APPEND IUT_LIST ${TARGET_IUTAVX2})
-endif(COMPILER_SUPPORTS_AVX2)
-
-if(COMPILER_SUPPORTS_AVX2)
-  add_executable(${TARGET_IUTAVX2128} iutsimd.c iutsimdmain.c testerutil.c)
-  target_compile_options(${TARGET_IUTAVX2128}
-    PRIVATE ${FLAGS_ENABLE_AVX2})
-  target_compile_definitions(${TARGET_IUTAVX2128}
-    PRIVATE ENABLE_AVX2128=1)
-  target_link_libraries(${TARGET_IUTAVX2128} ${LIB_SLEEF} ${EXTRA_LIBS})
-
-  list(APPEND IUT_LIST ${TARGET_IUTAVX2128})
-endif(COMPILER_SUPPORTS_AVX2)
-
-if(COMPILER_SUPPORTS_AVX512F)
-  add_executable(${TARGET_IUTAVX512F} iutsimd.c iutsimdmain.c testerutil.c)
-  target_compile_options(${TARGET_IUTAVX512F}
-    PRIVATE ${FLAGS_ENABLE_AVX512F})
-  target_compile_definitions(${TARGET_IUTAVX512F}
-    PRIVATE ENABLE_AVX512F=1)
-  target_link_libraries(${TARGET_IUTAVX512F} ${LIB_SLEEF} ${EXTRA_LIBS})
-
-  list(APPEND IUT_LIST ${TARGET_IUTAVX512F})
-endif(COMPILER_SUPPORTS_AVX512F)
-
-if(COMPILER_SUPPORTS_ADVSIMD)
-  # TODO: Remove relative-ness of the source file paths
-  add_executable(${TARGET_IUTADVSIMD} iutsimd.c iutsimdmain.c testerutil.c ../libm/sleefsimdsp.c ../libm/sleefsimddp.c)
-  target_compile_options(${TARGET_IUTADVSIMD}
-    PRIVATE ${FLAGS_ENABLE_ADVSIMD})
-  target_compile_definitions(${TARGET_IUTADVSIMD}
-    PRIVATE ENABLE_ADVSIMD=1)
-  target_link_libraries(${TARGET_IUTADVSIMD} ${LIB_SLEEF} ${EXTRA_LIBS})
-
-  list(APPEND IUT_LIST ${TARGET_IUTADVSIMD})
-endif(COMPILER_SUPPORTS_ADVSIMD)
+macro(test_extension SIMD)
+  if(COMPILER_SUPPORTS_${SIMD})
+    add_executable(${TARGET_IUT${SIMD}} ${IUT_SRC})
+    target_compile_options(${TARGET_IUT${SIMD}}
+      PRIVATE ${FLAGS_ENABLE_${SIMD}})
+    target_compile_definitions(${TARGET_IUT${SIMD}}
+      PRIVATE ENABLE_${SIMD}=1)
+    target_link_libraries(${TARGET_IUT${SIMD}} ${LIB_SLEEF} ${EXTRA_LIBS})
+
+    list(APPEND IUT_LIST ${TARGET_IUT${SIMD}})
+  endif(COMPILER_SUPPORTS_${SIMD})
+endmacro(test_extension)
+
+set(IUT_SRC iutsimd.c iutsimdmain.c testerutil)
+
+foreach(SIMD ${SLEEF_SUPPORTED_EXTENSIONS})
+  if(SIMD STREQUAL "ADVSIMD")
```

Comment:
Please build iutadvsimd like this for now, and create an issue saying that we need to build iutadvsimd like the other iut* executables.

Fixing this now would require changes in the c/h sources and in the makefiles that might break the makefiles builds. I'd rather leave it as it is.